pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -5520,3 +5520,164 @@ func TestRhdmProdImmutableEnvironmentWithoutJbpmCluster(t *testing.T) {
 	assert.Equal(t, int32(1), env.Servers[0].DeploymentConfigs[0].Spec.Replicas)
 	assert.Nil(t, cr.Status.Applied.Objects.Console, "Console should be nil")
 }
+
+func TestRhdmEnvironmentWithKafkaExt(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhdmProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						Kafka: createKafkaExtObject(),
+					},
+				},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.True(t, len(cr.Spec.Objects.Servers[0].Kafka.Topics) == 2)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	autocreateEnv, maxBlockMS, topics, extEnabled, clientID, acks, groupID, bootstrapServers := false, false, false, false, false, false, false, false
```

Comment:
what is the purpose of this line?

Reply:
Checks if all envs are present with the expected values

Reply:
I think there is a better way to test envs, see: 
https://github.com/kiegroup/kie-cloud-operator/blob/main/pkg/controller/kieapp/defaults/defaults_test.go#L406

Reply:
Ok